engine/src/SceneView.cpp

```diff
@@ -192,7 +248,23 @@ void SceneView::render()
 
     VertexArrayContext vaoContext(_fullScreenVao);
 
-    _scene->renderAllLights(_lightPosition, _lightBaseColor);
+    // Render all point lights
+    {
+      ProgramContext programContext(_pointLightingProgram);
+      _lightPosition = _pointLightingProgram.getUniform<glm::vec3>("lightPosition");
```

Comment:
You should get these uniforms once, and store them (probably above, where all the other uniforms are initialized). This also means you'll have to store separate versions of base color and cam position for the two shaders.

Also, make sure you actually set the value of the camera uniform before you actually call `renderAll[Directional|Point]Lights` on the scene. As of right now, it is only being set above right after getting the render context, which means the first time the uniform object is bound to nothing (not sure what the behavior is there), and all subsequent times it is only bound to the last bound shader (which in this case would be the directional one.